src/Swarm/Web.hs

```diff
@@ -70,16 +70,19 @@ defaultPort = 5357
 --   startup doesn't work.  Otherwise, ignore the failure.  In any
 --   case, return a @Maybe Port@ value representing whether a web
 --   server is actually running, and if so, what port it is on.
-startWebThread :: Maybe Warp.Port -> IORef GameState -> IO (Maybe Warp.Port)
+startWebThread :: Maybe Warp.Port -> IORef GameState -> IO (Either String Warp.Port)
 -- User explicitly provided port '0': don't run the web server
-startWebThread (Just 0) _ = pure Nothing
+startWebThread (Just 0) _ = pure $ Left "The web port has been turned off."
 startWebThread portM gsRef = do
   baton <- newEmptyMVar
   let port = fromMaybe defaultPort portM
   void $ forkIO $ webMain (Just baton) port gsRef
   res <- timeout 500_000 (takeMVar baton)
   case (portM, res) of
     -- User requested explicit port but server didn't start: fail
-    (Just _, Nothing) -> fail $ "Failed to start the web API  on :" <> show port
+    (Just _, Nothing) -> fail $ failMsg port
     -- Otherwise, just report whether the server is running, and if so, on what port
-    _ -> return (port <$ res)
+    (Just _, Just ()) -> return (Right port)
+    _ -> return . Left $ failMsg port
```

Comment:
This final catch-all case should not be a failure.  If the first value of the tuple is `Nothing` it just means that no explicit `--web` flag was given.  `(Nothing, Just ())` is definitely a success, that means the web server successfully started on the default port.

Actually, now that we have the Messages framework I think we can actually drop the whole thing about ignoring failure when the user didn't explicitly request a port.  In other words this `case (portM, res)` should just be `case res`, we don't care about `portM` at this point.

Reply:
Ups, my mistake! :sweat_smile:

Reply:
Well, I don't think we can entirely get rid of the `portM` in the `case`, in fact, I split it more.

But it should handle errors better now. This is what you get when you run two Swarms:

![image](https://user-images.githubusercontent.com/44544735/188430782-88ee82e5-0e4d-41c7-b9cc-b8fcaa00860d.png)